### PR TITLE
dev-util/rt-tests: A collection of latency testing tools (new package)

### PR DESCRIPTION
**Pull request by shifty91**
Hi,

rt-tests contains a set of programs that test and measure various components of
real-time kernel behavior, such as timer latency, signal latency and the
functioning of priority-inheritance mutexes. Additionally it contains hackbench,
a program to generate work for the scheduler.

This PR also replaces the older one: https://github.com/gentoo/gentoo/pull/11842

Bug: https://bugs.gentoo.org/705150
Signed-off-by: Kurt Kanzenbach <kurt@kmk-computers.de>

### CODE REVIEW

**Comment by shifty91** [comment]
Added the ebuild. Thanks!

File: dev-util/rt-tests/rt-tests-1.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Copyright 1999-2019 Gentoo Authors
```
Comment (juippis) [inline dev-util/rt-tests/rt-tests-1.5.ebuild]: Old copyright header, you can use repoman to commit so it fixes these things for you automatically.

File: dev-util/rt-tests/rt-tests-1.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=(python3_{5,6,7})
```
Comment (juippis) [inline dev-util/rt-tests/rt-tests-1.5.ebuild]: One empty line below EAPI.

Drop python3.5 support, it doesn't exist in tree anymore. If possible, test with 3.8 instead.

File: dev-util/rt-tests/rt-tests-1.5.ebuild [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+inherit python-single-r1
+
+DESCRIPTION="A collection of latency testing tools for the linux(-rt) kernel"
+HOMEPAGE="https://git.kernel.org/pub/scm/utils/rt-tests/rt-tests.git/about/"
+SRC_URI="https://kernel.org/pub/linux/utils/rt-tests/${P}.tar.xz"
+
+LICENSE="GPL-2 GPL-2+ LGPL-2.1+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~x86"
```
Comment (juippis) [inline dev-util/rt-tests/rt-tests-1.5.ebuild]: You can't add these keywords. You can only add amd64 and x86, provided you have **tested** this ebuild on these arches. You'll have to request keywording through bugzilla for any other arch.

File: dev-util/rt-tests/rt-tests-1.5.ebuild [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+IUSE="numa"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+DEPEND="${PYTHON_DEPS}
+	numa? ( sys-process/numactl )"
+RDEPEND="${DEPEND}"
+BDEPEND=""
```
Comment (juippis) [inline dev-util/rt-tests/rt-tests-1.5.ebuild]: Unused variable can be dropped.

File: dev-util/rt-tests/rt-tests-1.5.ebuild [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+PATCHES=( "${FILESDIR}/${P}-man-compression.patch" )
+
+src_compile() {
+	emake $(usex numa 'NUMA=1' 'NUMA=0') all
+}
+
+src_install() {
+	emake prefix=/usr DESTDIR="${D}" install
+	python_optimize
+}
```
Comment (juippis) [inline dev-util/rt-tests/rt-tests-1.5.ebuild]: Also remember to fix shebang on hwlatdetect.py or whatever this package installs.

File: dev-util/rt-tests/rt-tests-1.5.ebuild (hunk 3 above)
Comment (shifty91) [reply dev-util/rt-tests/rt-tests-1.5.ebuild]: I see. Thanks.

**Comment by shifty91** [comment]
@juippis Thanks for the review. Comments integrated.

File: dev-util/rt-tests/rt-tests-1.5.ebuild [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+EAPI=7
+
+PYTHON_COMPAT=(python3_{6,7,8})
+
+inherit python-single-r1
+
+DESCRIPTION="A collection of latency testing tools for the linux(-rt) kernel"
+HOMEPAGE="https://git.kernel.org/pub/scm/utils/rt-tests/rt-tests.git/about/"
+SRC_URI="https://kernel.org/pub/linux/utils/rt-tests/${P}.tar.xz"
```
Comment (juippis) [inline dev-util/rt-tests/rt-tests-1.5.ebuild]: I tried to test this, but seems like they dropped 1.5 altogether. Buggy release? Please update.

Comment (shifty91) [reply dev-util/rt-tests/rt-tests-1.5.ebuild]: It seems like that v1.6 has been releases in the meantime. I've updated it.
The old releases are still available on the server in the older/ directory.

**Pull request closed without merging**